src/theory/sep/theory_sep.cpp

```diff
@@ -1060,33 +1027,35 @@ int TheorySep::processAssertion(
   }
 
   if( !underSpatial && ( !references[index][n].empty() || card>0 ) ){
-    TypeNode tn = getReferenceType();
-    Assert(!tn.isNull());
+    Assert(!d_type_ref.isNull());
     // add references to overall type
-    unsigned bt = d_bound_kind[tn];
+    unsigned bt = d_bound_kind;
     bool add = true;
     if( references_strict[index].find( n )!=references_strict[index].end() ){
       Trace("sep-bound") << "Strict bound found based on " << n << ", index = " << index << std::endl;
       if( bt!=bound_strict ){
-        d_bound_kind[tn] = bound_strict;
-        //d_type_references[tn].clear();
-        d_card_max[tn] = card;
+        d_bound_kind = bound_strict;
+        d_card_max = card;
       }else{
         //TODO: derive static equality
         add = false;
       }
     }else{
       add = bt!=bound_strict;
     }
-    for( unsigned i=0; i<references[index][n].size(); i++ ){
-      if( std::find( d_type_references[tn].begin(), d_type_references[tn].end(), references[index][n][i] )==d_type_references[tn].end() ){
-        d_type_references[tn].push_back( references[index][n][i] );
+    for (const Node& r : references[index][n])
+    {
+      if (std::find(d_type_references.begin(), d_type_references.end(), r)
+          == d_type_references.end())
+      {
+        d_type_references.push_back(r);
       }
     }
     if( add ){
       //add max cardinality
-      if( card>(int)d_card_max[tn] ){
-        d_card_max[tn] = card;
+      if (card > (int)d_card_max)
```

Comment:
Can't card be declared as unsigned as well?

Reply:
Sure, I'll refactor.